pandas/tests/reshape/test_concat.py

```diff
@@ -1294,6 +1295,17 @@ def test_concat_rename_index(self):
         tm.assert_frame_equal(result, exp)
         assert result.index.names == exp.index.names
 
+    def test_concat_with_ordered_dict(self):
+        # GH 21510
+        result = pd.concat(OrderedDict([('First', pd.Series(range(3))),
+                                        ('Another', pd.Series(range(4)))]))
+        a = pd.Series(range(3), range(3))
```

Comment:
if you use the 2 form constructor, then specify ``index=``. I don't think it actually helps here though.

Reply:
Why should I specify that?